### PR TITLE
Fix openmp detection with clang-cl

### PR DESCRIPTION
**Pull request by peterjc123**

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -902,6 +902,32 @@ if(USE_OPENMP)
     ENDIF()
   ENDIF()
 
+  IF("${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC"
```
Comment (malfet) [inline cmake/Dependencies.cmake]: nit
```suggestion
  if("${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC"
```

File: cmake/Dependencies.cmake [hunk 2]
```diff
@@ -902,6 +902,32 @@ if(USE_OPENMP)
...
+    AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    MESSAGE(STATUS "Setting OpenMP flags for clang-cl")
+    SET(OpenMP_CXX_FLAGS "-Xclang -fopenmp")
+    SET(OpenMP_C_FLAGS "-Xclang -fopenmp")
+    SET(CHECKED_OPENMP ON CACHE BOOL "already checked for OpenMP")
+    SET(OPENMP_FOUND ON CACHE BOOL "OpenMP Support found")
+    if (NOT MKL_FOUND)
```
Comment (malfet) [inline cmake/Dependencies.cmake]: nit
```suggestion
    if(NOT MKL_FOUND)
```

File: cmake/Dependencies.cmake [hunk 3]
```diff
@@ -902,6 +902,32 @@ if(USE_OPENMP)
...
+        add_library(caffe2::openmp INTERFACE IMPORTED)
+      endif()
+
+      set_property(
+        TARGET caffe2::openmp PROPERTY INTERFACE_LINK_LIBRARIES
+        ${CLANG_OPENMP_LIBRARY})
+
+      LIST(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS caffe2::openmp)
+    endif()
+  ENDIF()
```
Comment (malfet) [inline cmake/Dependencies.cmake]: nit
```suggestion
  endif()
```

File: cmake/public/cuda.cmake [hunk 4]
```diff
@@ -435,6 +435,17 @@ else()
   list(APPEND CUDA_NVCC_FLAGS "-Xcompiler" "-fPIC")
 endif()
 
+# OpenMP flags for NVCC with Clang-cl
+if ("${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC"
```
Comment (malfet) [inline cmake/public/cuda.cmake]:
```suggestion
if("${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC"
```

File: cmake/public/cuda.cmake [hunk 5]
```diff
@@ -435,6 +435,17 @@ else()
...
+  AND "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+  list(APPEND CUDA_PROPAGATE_HOST_FLAGS_BLACKLIST "-Xclang" "-fopenmp")
+  if (MSVC_TOOLSET_VERSION LESS 142)
```
Comment (malfet) [inline cmake/public/cuda.cmake]:
```suggestion
  if(MSVC_TOOLSET_VERSION LESS 142)
```

**Review by malfet — APPROVED** [review]
Formatting (i.e. lowercase `if` with no space before opening brace ) otherwise LGTM

**Pull request closed without merging**